lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/MoreLists.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Collection;
+import java.util.List;
+
+public final class MoreLists
+{
+    private MoreLists() {}
+
+    public static <T> boolean containsAll(List<T> container, Collection<? extends T> searchedElements)
+    {
+        // A heuristic where it probably does not make sense to instantiate a set
+        if (container.size() <= 3 || searchedElements.size() <= 3) {
```

Comment:
What is this heuristic based on? How did you come up with it?

Reply:
I think we tried copying in some cases and there was a negative perf impact for some paths that mattered. If we're going to have a threshold, I would suggest running some microbenchmarks to see, approximately, where the cutoff should be.

Reply:
What kind of micro benchmarks would you suggest?

I believe benchmarks with narrow tables, like tpch, should not show this as a significant improvement, or maybe even not an improvement at all. Open to guidance.

Reply:
Any thoughts?

Reply:
Just a microbenchmarks for containsAll with target lists of different sizes. That should at least help inform the order of magnitude for that cutoff.